### PR TITLE
Move shared preferences to hilt

### PR DESCRIPTION
**Pull request by naivekook**
####  Description
In order to not depend on context everywhere and not to obtain SharedPreferences each time we trying to get language I moved it to hilt.

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/LocaleHelper.kt [hunk 1]
```diff
@@ -103,12 +104,12 @@ object LocaleHelper {
         return newContext
     }
 
-    fun setLanguageInPrefs(context: Context, language: String?): Context {
-        PreferenceManager.getDefaultSharedPreferences(context).edit {
+    fun setLanguageInPrefs(context: Context, language: String?, sharedPreferences: SharedPreferences): Context {
+        sharedPreferences.edit {
             putString(SELECTED_LANGUAGE, language)
             apply()
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/LocaleHelper.kt]:
```suggestion
```

`apply()` is useless when using `edit {}`

**Review by VaiTon — COMMENTED** [review]
Why do you want to move shared preferences to hilt when we have to pass context too?

**Comment by naivekook** [comment]
> Why do you want to move shared preferences to hilt when we have to pass context too?

this is the problem of current locale helper, my next PR will be refactor of LocaleHelper
I want move LocaleHelper in Hilt graph too and in this case you will not have to pass sharedprefs to it, because we will retrieve it from hilt in constructor

and because it will be in Hilt you will be able to use it in presenters and viemodels without having `context`

in code it will be like

```
@Inject
lateinit var localeHelper: LocaleHelper

val language = localeHelper.getLanguage()
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/LocaleHelper.kt (hunk 1 above)
Comment (naivekook) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/LocaleHelper.kt]: right!

**Comment by naivekook** [comment]
we can park this PR until I finish LocaleHelper refactor to see how it will look like

**Comment by VaiTon** [comment]
@naivekook please rebase so we can see if it pass tests

**Comment by VaiTon** [comment]
@naivekook rebase again please

**Comment by teolemon** [comment]
rebase needed @naivekook

**Comment by naivekook** [comment]
@VaiTon @teolemon rebased

**Review by VaiTon — COMMENTED** [review]
Why did you remove the rules from androidTesting?

**Comment by naivekook** [comment]
@VaiTon I didn't remove them , just moved to parent abstract class to make the hilt working automatically in child classes

**Pull request merged**